components/brave_wallet_ui/components/extension/connect-with-site-panel/select-account-item/select-account-item.tsx

```diff
@@ -0,0 +1,151 @@
+
+interface Props {
+  account: WalletAccountType
+  isSelected: boolean
+  onSelectAccount: () => void
+}
+
+export const SelectAccountItem = (props: Props) => {
+  const { account, isSelected, onSelectAccount } = props
+
+  // Wallet Selectors
+  const networks = useUnsafeWalletSelector(WalletSelectors.networkList)
+  const selectedNetwork = useUnsafeWalletSelector(
+    WalletSelectors.selectedNetwork
+  )
+  const userVisibleTokensInfo = useUnsafeWalletSelector(
+    WalletSelectors.userVisibleTokensInfo
+  )
+  const spotPrices = useUnsafeWalletSelector(
+    WalletSelectors.transactionSpotPrices
+  )
+  const defaultCurrencies = useUnsafeWalletSelector(
+    WalletSelectors.defaultCurrencies
+  )
+
+  // Memos
+  const orb = React.useMemo(() => {
+    return create({
+      seed: account.address.toLowerCase(),
+      size: 8,
+      scale: 16
+    }).toDataURL()
+  }, [account.address])
+
+  const tokenListByAccount = React.useMemo(() => {
+    if (
+      selectedNetwork &&
+      SupportedTestNetworks.includes(selectedNetwork?.chainId)
+    ) {
+      return userVisibleTokensInfo.filter(
+        (token) =>
+          token.visible &&
+          !token.isErc721 &&
+          !token.isNft &&
+          token.chainId === selectedNetwork.chainId &&
+          token.coin === selectedNetwork.coin
+      )
+    }
+    const chainList =
+      networks
+        .filter(
+          (network) =>
+            network.coin === account.coin &&
+            !SupportedTestNetworks.includes(network.chainId)
+        )
+        .map((network) => network.chainId) ?? []
+    return userVisibleTokensInfo.filter(
+      (token) =>
+        token.visible &&
+        !token.isErc721 &&
+        !token.isNft &&
+        chainList.includes(token.chainId)
+    )
+  }, [userVisibleTokensInfo, networks, account, selectedNetwork])
+
+  const accountFiatValue = React.useMemo(() => {
+    const amounts = tokenListByAccount.map((token) => {
+      const balance = getBalance(account, token)
+      return computeFiatAmount(spotPrices, {
+        decimals: token.decimals,
+        symbol: token.symbol,
+        value: balance,
+        contractAddress: token.contractAddress,
+        chainId: token.chainId
+      }).format()
+    })
+
+    if (amounts.length === 0) {
+      return ''
+    }
+
+    const reducedAmounts = amounts.reduce(function (a, b) {
+      return a !== '' && b !== '' ? new Amount(a).plus(b).format() : ''
+    })
+    return new Amount(reducedAmounts).formatAsFiat(defaultCurrencies.fiat)
+  }, [tokenListByAccount, spotPrices, defaultCurrencies.fiat])
+
+  return (
+    <ConnectPanelButton border="top" onClick={onSelectAccount}>
+      <LeftSide>
+        <AccountCircle orb={orb} />
+        <NameAndAddressColumn>
+          <AccountNameText>
+            {reduceAccountDisplayName(account.name, 22)}
+          </AccountNameText>
+          <AccountAddressText>
+            {reduceAddress(account.address)}
+          </AccountAddressText>
```

Comment:
Should this have a tooltip for showing the whole address or allowing to copy the address?

Reply:
Tooltips added!